docs/models_dev_guide.md

```diff
@@ -0,0 +1,263 @@
+
+### Model versions
+
+Each model is identified by a unique string of the form `<name>@<version>`,
+where `<name>` is the canonical name of the model and `<version>` is an
+optional version of the model.
+
+If a model is registered without a version, it can be referenced simply by
+its `<name>`. If a model is registered with a version, it can be referenced
+explicitly using the `<name>@<version>` syntax. The `<name>` syntax can
+also be used with versioned models, in which case the latest version of the
+model (i.e. the largest version number) will be referenced.
+
+#### Models search path
+
+The `models_dirs` field of the ETA config defines the list of directories to
+use when performing all models-related operations. If you add a new models
+directory to your path, you can use the `eta.core.models.init_models_dir`
+function to initialize it for use with the model management interface.
+
+#### Model manifest files
+
+Each directory on the model search path contains a `manifest.json` file that
+describes the models that have been registered in that directory. The manifest
+files are lightweight and are maintained in version control in the repository.
+
+The manifest file contains the names and versions of each model registered in
+its parent directory; it also contain the necessary information to locate
+each model in cloud storage. When a model is downloaded, it will be stored
+in the same directory as its corresponding manifest file.
+
+The following is an example contents of a `manifest.json` file:
+
+```json
+{
+    "models": [
+        {
+            "base_name": "VGG-16",
+            "base_filename": "vgg16_weights.npz",
+            "version": null,
+            "manager": {
+                "config": {
+                    "google_drive_id": "0B7phNvpRqNdpT0lZU1NOWXIzRlE"
+                },
+                "type": "eta.core.models.ETAModelManager"
+            },
+            "date_created": "2018-06-20 17:35:53"
+        },
+        ...
+    ]
+}
+```
+
+#### Model managers
+
+The `eta.core.models.ModelManager` class defines the base implementation of
+a model manager, which is a class that can upload, download, and delete a model
+from remote storage. Each entry in a manifest file contains a `manager` field
+that describes the model manager instance for that model.
+
+It is straightforward to implement subclasses of `ModelManager` to support
+custom cloud (or other remote) storage solutions based on your needs.
+
+All ETA models are currently stored in Google Drive in a publicly accessible
+folder. The `eta.core.models.ETAModelManager` class provides the underlying
+functionality used to download these models.
+
+> Note: For security reasons, the ETA model manager is unable to upload and
+> delete models. These tasks must be performed manually by a Voxel51
+> administrator.
+
+#### Model weights
+
+The `eta.core.models.ModelWeights` class defines the base implementation of
```

Comment:
Should we note that "weights" is ETA's term for any and all parameters or model coefficients?  EG Linear SVM coefficients would also be stored in these weights.

Reply:
Good point. I'll clarify.

Reply:
Done